File: code/backend/server_py/src/module_little_utils/do/todolist.py
from sqlmodel import Column, DateTime, Field, SQLModel
from uuid import uuid4
from datetime import datetime, timezone
from enum import Enum


# 状态枚举
class TodoStatus(str, Enum):
    Todo = "todo"
    Done = "done"
    # 其他状态...
    # 暂停
    Pause = "pause"


class TodolistBase(SQLModel):
    """
    计划列表基础模型(不含数据库表配置)
    """

    pid: str | None = Field(None, description="父级ID")
    name: str | None = Field(max_length=100, description="计划列表名称")
    value: str = Field(default="", description="内容")

    description: str | None = Field(
        default=None, max_length=500, description="计划列表描述"
    )
    start_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        sa_column=Column(DateTime(timezone=True)),
        description="开始",
    )
    end_at: datetime | None = Field(
        default=None, sa_column=Column(DateTime(timezone=True)), description="结束"
    )

    status: TodoStatus = Field(default=TodoStatus.Todo, description="代办状态")


class Todolist(TodolistBase, table=True):
    """
    计划列表数据库模型(对应数据库表)
    """

    id: str = Field(
        default_factory=lambda: uuid4().hex,
        primary_key=True,  # 主键
        index=True,  # 索引
        description="唯一标识符",
    )
    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        sa_column=Column(DateTime(timezone=True)),
        description="创建时间",
    )
    updated_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        sa_column=Column(
            DateTime(timezone=True),
            onupdate=lambda: datetime.now(timezone.utc),  # 自动更新
            nullable=False,  # 不允许为空
        ),
        description="最后更新时间",
    )


class TodolistCreate(TodolistBase):
    """
    创建计划列表的请求模型
    """

    pass


class TodolistUpdate(TodolistBase):
    """
    更新计划列表的请求模型
    """

    updated_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        sa_column=Column(
            DateTime(timezone=True),
            onupdate=lambda: datetime.now(timezone.utc),  # 自动更新
            nullable=False,  # 不允许为空
        ),
        description="最后更新时间",
    )

File: code/frontend/vue3/src/modules/site/types/todolist.ts
// src/modules/site/types/todolist.ts —— 备忘类型(承接原 little_utils/todolist)

/** 备忘状态: 待办 / 完成 / 暂停 */
export type TodoStatus = 'todo' | 'done' | 'pause'

/** 备忘实体(start_at 为日历展示时间) */
export interface Todolist {
  id: string
  pid?: string | null
  /** 备忘标题(日历展示) */
  name: string | null
  /** 备忘全文内容(周视图取前200字) */
  value: string
  description?: string | null
  /** 备忘时间(ISO, 日历归属依据) */
  start_at: string
  end_at?: string | null
  status: TodoStatus
  user_id?: string | null
  created_at: string
  updated_at: string
}

/** 创建备忘参数 */
export interface TodolistCreate {
  name: string
  value?: string
  description?: string | null
  /** 备忘时间(ISO), 缺省为当前时间 */
  start_at?: string
  end_at?: string | null
  status?: TodoStatus
}

/** 更新备忘参数(全部可选, 仅更新传入字段) */
export type TodolistUpdate = Partial<TodolistCreate>
